pkg/service/endpoints.go

```diff
@@ -71,7 +71,7 @@ func (s *Service) updateHAProxySrv(client api.HAProxyClient, srvSlot store.HAPro
 		srv.Maintenance = "disabled"
 	}
 	// Cookie/Session persistence
-	if s.backend.Cookie != nil && s.backend.Cookie.Type == "insert" {
+	if s.backend.Cookie != nil && s.backend.Cookie.Type == "insert" && !s.backend.Cookie.Dynamic {
```

Comment:
When `cookie-persistence` is used, backends will always use dynamic cookies:

https://github.com/haproxytech/kubernetes-ingress/blob/e3b7ce1b1c17bc97ff894d3da8c5df97f1ecb19b/pkg/annotations/service/cookie.go#L33-L40

I think we should remove this `if` and never set `srv.Cookie`

Reply:
sorry for the delay. I have updated to just remove the setting of srv.cookie